apps/dashr-financial-report/assets/FinancialFunctions.R

```diff
@@ -0,0 +1,130 @@
+# Install a package that is necessary and has not been installed, or load one that is.
+
+usePackage <- function(p) 
+{
+  if (!is.element(p, installed.packages()[,1]))
+    install.packages(p, dep = TRUE)
+  require(p, character.only = TRUE)
+}
+
+
+# Generate a Table Function 
+generate_table <- function(df, nrows=20)
+  
+  # function generates a dash table from a supplied data frame (df)
+  
+  #  and number of rows (nrows) to display
+{
+  
+  n <- min(nrows, nrow(df))
+  
+  rows <- lapply(seq(1, n), function(i) {
+    
+    htmlTr(children = lapply(as.character(df[i,]), htmlTd))
+    
+  })
+  
+  header <- htmlTr(children = lapply(names(df), htmlTh))
+  
+  htmlTable(
+    
+    children = c(list(header), rows)
+  )
+}
+
+# Remove.Factors
+
+remove.factors <- function (df) 
+{
+  for (varnum in 1:length(df)) {
+    if ("factor" %in% class(df[, varnum])) {
+      df[varnum] = as.character(df[, varnum])
+    }
+  }
+  return(df)
+}
+
+
+
+#Create a HTML Header:
+
+Header <- function() {
+  return(
+    htmlDiv(list(
+      getLogo(),
+      htmlBr(),
+      getHeader(),
+      htmlDiv(list(
+        grey_line
+      )),
+      htmlBr(list()),

```

Comment:
Hmm. Does this require an empty list be passed? What happens when it's invoked with no arguments?

Reply:
It does not. Fixed.